### ISSUE TITLE
Issue when using `bloom-filter-scala` on String

### ISSUE THREAD

**Issue by wgx731** [alexandrnikitin/bloom-filter-scala#29]
@alexandrnikitin,

Awesome work on `bloom-filter-scala`, I am using `bloom-filter-scala` on `String`, but seems like false positive is too high. My demo example as below:
```
package sample

import bloomfilter.mutable.BloomFilter

object Sample extends App {

  // scalastyle:off
  println("""val bf = BloomFilter[String](10000, 0.1)""")
  val bf = BloomFilter[String](10000, 0.1)
  println("""bf.add("a") """)
  bf.add("a")
  println(s"""bf.mightContain("s"): ${bf.mightContain("s")}""")
  println(s"""bf.mightContain("1"): ${bf.mightContain("1")}""")
  println(s"""bf.mightContain(""): ${bf.mightContain("")}""")
  // scalastyle:on

}
```
Expecting all return `false`, but seems always return at least 2 `true`. May I know if I am using `bloom-filter-scala` correctly?

Thank you very much.

**Comment by alexandrnikitin** [comment]
@wgx731 Thanks for the issue. It seems the problem is in how we extract data from strings: https://github.com/alexandrnikitin/bloom-filter-scala/blob/471fccf346d4505b3551b45c0cb3616de2c402a3/bloom-filter/src/main/scala/bloomfilter/CanGetDataFrom.scala#L30-L39
It affects short strings. Do you want to take a look at the issue? You can take https://github.com/OpenHFT/Zero-Allocation-Hashing as an example.

**Comment by wgx731** [comment]
@alexandrnikitin, thx for the quick feedback. I am not algorithm expert and still studying bloom filter,  but I will try and see if I can help. Thanks again.

**Comment by alexandrnikitin** [comment]
@wgx731 I fixed the issue in `v0.10.0`. Please take a look.
